### ISSUE TITLE
Drizzle ORM Adapter with Github provider error

### ISSUE THREAD

**Issue by HugeLetters** [nextauthjs/next-auth#8349]
### Environment

```bash
  System:
    OS: Windows 10 10.0.22621
    CPU: (16) x64 AMD Ryzen 7 5700U with Radeon Graphics
    Memory: 2.12 GB / 15.33 GB
  Binaries:
    Node: 20.1.0 - C:\Program Files\nodejs\node.EXE
    npm: 9.8.1 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.6.12 - ~\AppData\Local\pnpm\pnpm.CMD
  Browsers:
    Edge: Spartan (44.22621.2134.0), Chromium (115.0.1901.203)
    Internet Explorer: 11.0.22621.1
```

### Reproduction URL

https://github.com/HugeLetters/recallio/tree/oauth-providers

### Describe the issue

Docs mention that GitHub provides an extra field for the account schema `refresh_token_expires_in` and that your DB schema should update accordingly. 
I'm using MySQL with Planetscale but I would assume the problem persists with any RDBMS. The problem is that adapter does not rely on my defined tables at all and instead just creates them like [so](https://github.com/nextauthjs/next-auth/blob/28e4328704776fcf1c88003b3eb6e5c28d15d49c/packages/adapter-drizzle/src/lib/mysql.ts#L14C19-L14C19]) which works most of the time but the problem is that [here](https://github.com/nextauthjs/next-auth/blob/28e4328704776fcf1c88003b3eb6e5c28d15d49c/packages/adapter-drizzle/src/lib/mysql.ts#L166C15-L166C15) a raw account object is passed which in case of GitHub will have properties not defined in your adapter schema(but defined in my schema which is not used here) and Drizlle will crash trying to map a property to column which doesn't exist in schema mapping.

Essentially this.
```ts
export const user = mysqlTable("user", {
  id: int('id'),
}
const value={id: '12345'; email: 'JohnDoe@gmail.com'};
// because TS checks for excessive properties only on object literals
// this will pass w/o problems and crash at runtime because Drizzle will try to find a column for email and would fail
await db.insert(user).values(value)
```

The way to easily test that this is the root cause is to add `delete rawAccount.refresh_token_expires_in` to `linkAccount` method before DB insert. 

I would make an issue in Drizzle repo but I think the problem in that case is there doesn't seem to be a simple mechanism for me to provide my schemas to next-auth except for the one I describe below.

### How to reproduce

1. Clone the repo
2. Set up either a Planetscale database or connect Drizzle to your own.
3. Run a migration
4. Run the dev server and try to login with GitHub with the button below

### Expected behavior

1. Either this extra field `refresh_token_expires_in` is not required at all - I couldn't find it being used in your codebase but perhaps the intention is for me have a use case for such data. If it's actually not needed then perhaps the best solution would be to strip this field before linking an account - but as far as I understand there's no way to do this with just the code for GitHub provider and so may not be favorable. 

2. The other option would be to allow to pass in my schema definition to Drizzle adapter explicitly and then use these tables instead of default ones if provided. 

3. This could also be solved by providing a custom table function like this but feels kinda hacky but if that's what you intended for people to do then ok and I guess the issue should be closed as wontfix.
```ts
import { account, session, user, verificationToken } from "@/database/schema";
const TABLES = {
  user,
  account,
  session,
  verificationToken,
};
function getTableByName(tableName: keyof typeof TABLES) {
  return TABLES[tableName];
}
export const authOptions: NextAuthOptions = {
  adapter: DrizzleAdapter(db, getTableByName as unknown as MySqlTableFn),
}
```

### I would gladly issue a PR if you would like my help.

**Comment by ArjobanSingh** [comment]
Even I found the same issue. To overcome this I created custom drizzle adapter and used the tables defined in my schema. It's really weird the package is creating schema internally instead of asking as param from the developer.

**Comment by HugeLetters** [comment]
Yup, I've also just copied default Drizzle adapter and created my custom one from that
